### DISCUSSION TITLE
Rationale behind enforcing TLS

### DISCUSSION THREAD

**Discussion by martino87r** [keycloak/keycloak#33588] [Q&A]
I'm running a keycloak instance behind a loadbalancer (a very basic HA config), and was trying for a few hours to get rid of TLS since It's terminated at the LB level. The LB manages the certificates automatically via ACME so basically with any application in my environment i don't deal with certificates (simplifies a lot management).

Now here I am trying to deploy keycloak only to be greeted by:

```
sh-5.2$ ./bin/kc.sh start 
Key material not provided to setup HTTPS. Please configure your keys/certificates or start the server in development mode.
```

So my only option is to run keycloak in production with the `dev-mode` on if I don't want to provide it with certs?

**Comment by shawkins** [comment]
> So my only option is to run keycloak in production with the dev-mode on if I don't want to provide it with certs?

That message should also include a reference to http-enabled, which you can use in tls termination scenario. Can you create an issue from this discussion for this to be addressed?

**Reply by martino87r** [reply]
I'm not sure what do you mean by the "reference to http-enabled". If I'm seeing options to either enable or disable a protocol I'm assuming the relative stuff should be too; so I was expecting something as simple as `https-enable` to be implemented, but it's not, and judging by the amount of people asking for it (since most people run it in containers) I'm quite surprised.
My conclusion here was that:
```
dev-server !https unless explicitly set -> http only by default
server -> https implcitly required without options to disable it
```

**Reply by martino87r** [reply]
Now I see the problem here:

if http-enabled is true and certstores or certpaths are not defined then only http mode is enabled;
in my config i omitted `http-enabled` because I assumed it's true by default.

**Reply by shawkins** [reply]
If you think it will help to have a reference to http-enabled from that initial error message, please create an issue.
